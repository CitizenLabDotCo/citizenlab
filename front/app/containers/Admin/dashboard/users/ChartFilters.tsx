import React from 'react';

// components
import { Box } from '@citizenlab/cl2-component-library';
import GroupFilter from '../components/filters/GroupFilter';
import TimeControl from '../components/TimeControl';

// typings
import { IOption } from 'typings';
import { Moment } from 'moment';

interface Props {
  startAtMoment?: Moment | null | undefined;
  endAtMoment: Moment | null;
  currentGroupFilter?: string;
  onChangeTimeRange: (
    startAtMoment: Moment | null,
    endAtMoment: Moment | null
  ) => void;
  onGroupFilter: (filter: IOption) => void;
}

const ChartFilters = ({
  startAtMoment,
  endAtMoment,
  currentGroupFilter,
  onGroupFilter,
  onChangeTimeRange,
}: Props) => {
  return (
    <Box width="100%" display="flex" mt="-10px">
      <TimeControl
        startAtMoment={startAtMoment}
        endAtMoment={endAtMoment}
        onChange={onChangeTimeRange}
      />
      <Box ml="12px">
        <GroupFilter
          currentGroupFilter={currentGroupFilter}
          onGroupFilter={onGroupFilter}
        />
      </Box>
    </Box>
  );
};

export default ChartFilters;
