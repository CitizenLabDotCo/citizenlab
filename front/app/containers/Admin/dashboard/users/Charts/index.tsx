import React, { PureComponent } from 'react';

// services
import {
  usersByRegFieldStream,
  IUsersByRegistrationField,
  usersByRegFieldXlsxEndpoint,
} from 'services/userCustomFieldStats';

// hooks
import useUserCustomFields from 'hooks/useUserCustomFields';

// intl
import { injectIntl } from 'utils/cl-intl';
import { WrappedComponentProps } from 'react-intl';
import localize, { InjectedLocalized } from 'utils/localize';
import messages from 'containers/Admin/dashboard/messages';

// components
import BarChartByCategory from './BarChartByCategory';
import PieChartByCategory from './PieChartByCategory';
import AreaChart from './AreaChart';
import GenderChart from './GenderChart';
import AgeChart from './AgeChart';
import { Box } from '@citizenlab/cl2-component-library';

// utils
import { isNilOrError, NilOrError } from 'utils/helperUtils';

// typings
import {
  IUserCustomFieldInputType,
  IUserCustomFieldData,
} from 'services/userCustomFields';

interface InputProps {
  currentGroupFilter: string | undefined;
  currentGroupFilterLabel: string | undefined;
  startAt: string | null | undefined;
  endAt: string | null;
}

interface DataProps {
  customFields: IUserCustomFieldData[] | NilOrError;
}

type GraphOption = {
  value: number;
  name: string;
  code: string;
};

export interface Props extends InputProps, DataProps {}

export class RegistrationFieldsToGraphs extends PureComponent<
  Props & WrappedComponentProps & InjectedLocalized
> {
  convertToGraphFormat = (data: IUsersByRegistrationField) => {
    const {
      series: { users },
      options,
    } = data;
    let res: GraphOption[] = [];
    if (options) {
      res = Object.entries(options)
        .sort((a, b) => a[1].ordering - b[1].ordering)
        .map(([key, value]) => ({
          value: users[key] || 0,
          name: this.props.localize(value.title_multiloc),
          code: key,
        }));
    }

    if (users['_blank']) {
      res.push({
        value: users['_blank'],
        name: this.props.intl.formatMessage(messages._blank),
        code: '_blank',
      });
    }

    return res.length > 0 ? res : null;
  };
  convertCheckboxToGraphFormat = (data: IUsersByRegistrationField) => {
    const {
      series: { users },
    } = data;
    const res = ['_blank', 'true', 'false'].map((key) => ({
      value: users[key] || 0,
      name: this.props.intl.formatMessage(messages[key]),
      code: 'key',
    }));

    return res.length > 0 ? res : null;
  };

  render() {
    const {
      customFields,
      localize,
      startAt,
      endAt,
      currentGroupFilter,
      currentGroupFilterLabel,
    } = this.props;

    if (isNilOrError(customFields)) {
      return null;
    }

    return customFields.map((field, index) => {
      if (field.attributes.enabled) {
        if (field.attributes.code === 'birthyear') {
          return (
            <Box width="50%" key={index}>
              <AgeChart
                startAt={startAt}
                endAt={endAt}
                currentGroupFilter={currentGroupFilter}
                currentGroupFilterLabel={currentGroupFilterLabel}
              />
            </Box>
          );
        }
        if (field.attributes.input_type === 'number') {
          return;
        }
        if (field.attributes.code === 'gender') {
          return (
            <Box width="50%" key={index}>
              <GenderChart
                startAt={startAt}
                endAt={endAt}
                currentGroupFilter={currentGroupFilter}
                currentGroupFilterLabel={currentGroupFilterLabel}
              />
            </Box>
          );
        }

        if (field.attributes.code === 'domicile') {
          return (
            <AreaChart
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
            />
          );
        }

        if (field.attributes.input_type === 'checkbox') {
          return (
            <PieChartByCategory
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
              convertToGraphFormat={this.convertCheckboxToGraphFormat}
              graphTitleString={localize(field.attributes.title_multiloc)}
              stream={usersByRegFieldStream}
              graphUnit="users"
              customId={field.id}
              xlsxEndpoint={usersByRegFieldXlsxEndpoint(field.id)}
            />
          );
        } else {
          return (
            <BarChartByCategory
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
              convertToGraphFormat={this.convertToGraphFormat}
              graphTitleString={localize(field.attributes.title_multiloc)}
              stream={usersByRegFieldStream}
              graphUnit="users"
              customId={field.id}
              xlsxEndpoint={usersByRegFieldXlsxEndpoint(field.id)}
            />
          );
        }
      }

      return null;
    });
  }
}

const RegistrationFieldsToGraphsWithHoCs = localize<Props>(
  injectIntl(RegistrationFieldsToGraphs as any)
) as any;

const INPUT_TYPES: IUserCustomFieldInputType[] = [
  'select',
  'multiselect',
  'checkbox',
  'number',
];

export default (inputProps: InputProps) => {
  const userCustomFields = useUserCustomFields({
    inputTypes: INPUT_TYPES,
  });

  return (
    <RegistrationFieldsToGraphsWithHoCs
      {...inputProps}
      customFields={userCustomFields}
    />
  );
};
