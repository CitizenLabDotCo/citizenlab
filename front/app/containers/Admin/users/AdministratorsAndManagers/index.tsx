import React, { useState } from 'react';
import { isString, isEmpty } from 'lodash-es';

import UserManager from '../UserManager';
import UsersHeader from '../UsersHeader';

import messages from '../messages';

const AllUsers = () => {
  const [search, setSearch] = useState<string | undefined>(undefined);

  const searchUser = (searchTerm: string) => {
    setSearch(isString(searchTerm) && !isEmpty(searchTerm) ? searchTerm : '');
  };

  return (
    <>
      <UsersHeader
        onSearch={searchUser}
        title={messages.administratorsAndManagers}
        subtitle={messages.administratorsAndManagersSubtitle}
      />
      <UserManager search={search} canModerate notCitizenlabMember />
    </>
  );
};

export default AllUsers;
