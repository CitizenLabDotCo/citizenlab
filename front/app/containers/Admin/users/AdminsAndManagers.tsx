import React, { useState } from 'react';
import { isString, isEmpty } from 'lodash-es';

import UserManager from './UserManager';
import UsersHeader from './UsersHeader';
import SeatInfo from 'components/SeatInfo';
import { Box } from '@citizenlab/cl2-component-library';
import messages from './messages';
import styled from 'styled-components';

const StyledBox = styled(Box)`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-auto-rows: 1fr;
`;

const AllUsers = () => {
  const [search, setSearch] = useState<string | undefined>(undefined);

  const searchUser = (searchTerm: string) => {
    setSearch(isString(searchTerm) && !isEmpty(searchTerm) ? searchTerm : '');
  };

  return (
    <>
      <UsersHeader
        onSearch={searchUser}
        title={messages.adminsAndManagers}
        subtitle={messages.adminsAndManagersSubtitle}
      />
      <UserManager search={search} canModerate notCitizenlabMember />
      <StyledBox>
        <SeatInfo seatType="admin" width={null} />
        <SeatInfo seatType="project_manager" width={null} />
      </StyledBox>
    </>
  );
};

export default AllUsers;
