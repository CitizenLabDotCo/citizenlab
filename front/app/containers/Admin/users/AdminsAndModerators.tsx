import React, { useState } from 'react';
import { isString, isEmpty } from 'lodash-es';

import UserManager from './UserManager';
import UsersHeader from './UsersHeader';
import SeatInfo from 'components/SeatBasedBilling/SeatInfo';
import { Box } from '@citizenlab/cl2-component-library';
import messages from './messages';
import styled from 'styled-components';

const StyledBox = styled(Box)`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-auto-rows: 1fr;
`;

const AdminsAndModerators = () => {
  const [search, setSearch] = useState<string | undefined>(undefined);

  const searchUser = (searchTerm: string) => {
    setSearch(isString(searchTerm) && !isEmpty(searchTerm) ? searchTerm : '');
  };

  return (
    <>
      <UsersHeader
        onSearch={searchUser}
        title={messages.adminsAndManagers}
        subtitle={messages.adminsAndManagersSubtitle}
      />
      <UserManager
        search={search}
        canModerate
        notCitizenlabMember
        includeInactive
      />
      <StyledBox mt="20px">
        <SeatInfo seatType="admin" />
        <SeatInfo seatType="moderator" />
      </StyledBox>
    </>
  );
};

export default AdminsAndModerators;
