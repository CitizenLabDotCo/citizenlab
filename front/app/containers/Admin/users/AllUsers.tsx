// Libraries
import React, { PureComponent } from 'react';
import { isString, isEmpty } from 'lodash-es';

// Components
import UserManager from './UserManager';
import UsersHeader from './UsersHeader';

// Styles
import styled from 'styled-components';

const Container = styled.div``;

interface Props {}

interface State {
  search: string | undefined;
}

export default class AllUsers extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      search: undefined,
    };
  }

  searchUser = (searchTerm: string) => {
    this.setState({
      search: isString(searchTerm) && !isEmpty(searchTerm) ? searchTerm : '',
    });
  };

  render() {
    const { search } = this.state;

    return (
      <Container>
        <UsersHeader onSearch={this.searchUser} />
        <UserManager search={search} />
      </Container>
    );
  }
}
