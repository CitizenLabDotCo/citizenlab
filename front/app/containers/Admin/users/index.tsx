import React, { PureComponent } from 'react';
import { withRouter, WithRouterProps } from 'utils/cl-router/withRouter';
import { Outlet as RouterOutlet } from 'react-router-dom';

// Resources

// components
import HelmetIntl from 'components/HelmetIntl';
import Modal from 'components/UI/Modal';
import GroupsListPanel from './GroupsListPanel';
import GroupCreationStep1 from './GroupCreationStep1';
import NormalGroupForm, { NormalFormValues } from './NormalGroupForm';

// Global state
import {
  globalState,
  IAdminNoPadding,
  IGlobalStateService,
} from 'services/globalState';

// Styling
import styled from 'styled-components';
import { media } from 'utils/styleUtils';

const Wrapper = styled.div`
  height: calc(100vh - ${(props) => props.theme.menuHeight}px);
  display: flex;
  background: #fff;
  position: fixed;
  right: 0;
  top: ${(props) => props.theme.menuHeight}px;
  left: 210px;
  bottom: 0;
  ${media.tablet`
    left: 80px;
  `}
`;

const LeftPanel = styled(GroupsListPanel)`
  width: 300px;
  flex: 0 0 300px;

  ${media.tablet`
    width: 260px;
    flex: 0 0 260px;
  `}
`;

const ChildWrapper = styled.div`
  flex: 1;
  padding: 50px;
  background: white;
  overflow-x: auto;
  overflow-y: auto;
`;

// i18n
import FormattedMessage from 'utils/cl-intl/FormattedMessage';
import messages from './messages';

// Services
import { IGroupData, addGroup, MembershipType } from 'services/groups';

export interface Props {}

export type GroupCreationModal = false | 'step1' | MembershipType;

export interface State {
  groupCreationModal: GroupCreationModal;
}

class UsersPage extends PureComponent<Props & WithRouterProps, State> {
  globalState: IGlobalStateService<IAdminNoPadding>;

  constructor(props: Props & WithRouterProps) {
    super(props);
    this.globalState = globalState.init('AdminNoPadding', { enabled: true });
    this.state = {
      groupCreationModal: false,
    };
  }

  componentDidMount() {
    this.globalState.set({ enabled: true });
  }

  componentWillUnmount() {
    this.globalState.set({ enabled: false });
  }

  openGroupCreationModal = () => {
    this.setState({ groupCreationModal: 'step1' });
  };

  closeGroupCreationModal = () => {
    this.setState({ groupCreationModal: false });
  };

  openStep2 = (groupType: IGroupData['attributes']['membership_type']) => {
    this.setState({ groupCreationModal: groupType });
  };

  handleSubmitForm = async (values: NormalFormValues) => {
    await addGroup({ ...values });
    this.closeGroupCreationModal();
  };

  renderModalHeader = () => {
    return <FormattedMessage {...messages.modalHeaderStep1} />;
  };

  render() {
    if (!this.props.location) return null;

    const { groupCreationModal } = this.state;

    return (
      <>
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />

        <Wrapper>
          <LeftPanel
            className="e2e-left-panel"
            onCreateGroup={this.openGroupCreationModal}
          />
          <ChildWrapper id="e2e-users-container">
            <RouterOutlet />
          </ChildWrapper>
        </Wrapper>

        <Modal
          header={this.renderModalHeader()}
          opened={groupCreationModal !== false}
          close={this.closeGroupCreationModal}
        >
          <>
            {groupCreationModal === 'step1' && (
              <GroupCreationStep1 onOpenStep2={this.openStep2} />
            )}

            {groupCreationModal === 'manual' && (
              <NormalGroupForm
                defaultValues={{
                  membership_type: 'manual',
                }}
                onSubmit={this.handleSubmitForm}
              />
            )}
          </>
        </Modal>
      </>
    );
  }
}

const UsersPageWithHocs = withRouter(UsersPage);

export default UsersPageWithHocs;
