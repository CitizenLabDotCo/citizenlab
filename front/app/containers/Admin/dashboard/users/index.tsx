// libraries
import React, { PureComponent } from 'react';
import moment, { Moment } from 'moment';
import { adopt } from 'react-adopt';
import localize, { InjectedLocalized } from 'utils/localize';

// resources
import GetGroups, { GetGroupsChildProps } from 'resources/GetGroups';

// components
import ChartFilters from '../components/ChartFilters';
import { GraphsContainer, ControlBar } from 'components/admin/Chart';
import TimeControl from '../components/TimeControl';
import MostActiveUsersList from './charts/MostActiveUsersList';
import Outlet from 'components/Outlet';

// i18n
import { injectIntl } from 'utils/cl-intl';
import messages from '../messages';
import { InjectedIntlProps } from 'react-intl';

// tracking
import { injectTracks } from 'utils/analytics';
import tracks from '../tracks';

// typings
import { IOption } from 'typings';

// utils
import { isNilOrError } from 'utils/helperUtils';

interface State {
  startAtMoment?: Moment | null;
  endAtMoment: Moment | null;
  currentGroupFilter: string | undefined;
  currentGroupFilterLabel: string | undefined;
}

interface DataProps {
  groups: GetGroupsChildProps;
}

interface Props extends DataProps {}

interface Tracks {
  trackFilterOnGroup: (args: { extra: Record<string, string> }) => void;
}

export class UsersDashboard extends PureComponent<
  Props & InjectedIntlProps & InjectedLocalized & Tracks,
  State
> {
  constructor(props: Props & InjectedIntlProps & InjectedLocalized & Tracks) {
    super(props as any);
    this.state = {
      startAtMoment: undefined,
      endAtMoment: moment(),
      currentGroupFilter: undefined,
      currentGroupFilterLabel: undefined,
    };
  }

  handleChangeTimeRange = (
    startAtMoment: Moment | null | undefined,
    endAtMoment: Moment | null
  ) => {
    this.setState({ startAtMoment, endAtMoment });
  };

  handleOnGroupFilter = (filter) => {
    this.props.trackFilterOnGroup({ extra: { group: filter } });
    this.setState({
      currentGroupFilter: filter.value,
      currentGroupFilterLabel: filter.label,
    });
  };

  generateGroupFilterOptions = () => {
    const {
      groups,
      groups: { groupsList },
      localize,
      intl: { formatMessage },
    } = this.props;
    let filterOptions: IOption[] = [];

    if (!isNilOrError(groups) && !isNilOrError(groupsList)) {
      filterOptions = groupsList.map((group) => ({
        value: group.id,
        label: localize(group.attributes.title_multiloc),
      }));
    }

    return [
      { value: '', label: formatMessage(messages.allGroups) },
      ...filterOptions,
    ];
  };

  render() {
    const {
      currentGroupFilter,
      endAtMoment,
      startAtMoment,
      currentGroupFilterLabel,
    } = this.state;
    const startAt = startAtMoment && startAtMoment.toISOString();
    const endAt = endAtMoment && endAtMoment.toISOString();
    const infoMessage = this.props.intl.formatMessage(
      messages.mostActiveUsersRankingDescription
    );

    return (
      <>
        <ControlBar>
          <TimeControl
            startAtMoment={startAtMoment}
            endAtMoment={endAtMoment}
            onChange={this.handleChangeTimeRange}
          />
        </ControlBar>

        <ChartFilters
          currentProjectFilter={undefined}
          currentGroupFilter={currentGroupFilter}
          currentTopicFilter={undefined}
          projectFilterOptions={null}
          groupFilterOptions={this.generateGroupFilterOptions()}
          topicFilterOptions={null}
          onProjectFilter={null}
          onGroupFilter={this.handleOnGroupFilter}
          onTopicFilter={null}
        />

        <GraphsContainer>
          <Outlet
            id="app.containers.Admin.dashboard.users.graphs"
            startAt={startAt}
            endAt={endAt}
            currentGroupFilter={currentGroupFilter}
            currentGroupFilterLabel={currentGroupFilterLabel}
          />
          <MostActiveUsersList
            currentGroupFilter={currentGroupFilter}
            startAt={startAt}
            endAt={endAt}
            infoMessage={infoMessage}
            className="dynamicHeight"
          />
        </GraphsContainer>
      </>
    );
  }
}

const Data = adopt<DataProps>({
  groups: <GetGroups />,
});

const UsersDashBoardWithHOCs = injectIntl(
  injectTracks<Props>({
    trackFilterOnGroup: tracks.filteredOnGroup,
  })(localize<Props & Tracks>(UsersDashboard))
);

export default () => (
  <Data>{(dataProps) => <UsersDashBoardWithHOCs {...dataProps} />}</Data>
);
