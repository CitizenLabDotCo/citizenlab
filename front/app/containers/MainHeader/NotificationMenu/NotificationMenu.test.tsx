// libraries
import React from 'react';
import { shallow } from 'enzyme';

// component to test
import { NotificationMenu } from './';
import { GetNotificationsChildProps } from 'resources/GetNotifications';
import { GetAuthUserChildProps } from 'resources/GetAuthUser';

// mock utilities
jest.mock('utils/cl-intl');
jest.mock('utils/analytics', () => ({ trackEventByName: () => {} }));

import { getNotification } from 'services/__mocks__/notifications';
import { makeUser } from 'services/__mocks__/users';
jest.mock('modules', () => ({ streamsToReset: [] }));

const onLoadMore = jest.fn();
const mockNotificationsFromResourceComp: GetNotificationsChildProps = {
  onLoadMore,
  list: [getNotification('aminRights', 'admin_rights_received')],
  hasMore: false,
};
const mockUserFromResource: GetAuthUserChildProps = makeUser({
  unread_notifications: 0,
}).data;

describe('<NotificationMenu />', () => {
  it('renders correctly when everything is there', () => {
    const wrapper = shallow(
      <NotificationMenu
        notifications={mockNotificationsFromResourceComp}
        authUser={mockUserFromResource}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when there is no user', () => {
    const wrapper = shallow(
      <NotificationMenu
        notifications={mockNotificationsFromResourceComp}
        authUser={undefined}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when there is no notifs', () => {
    const noNotifs = {
      onLoadMore,
      list: null,
      hasMore: false,
    };
    const wrapper = shallow(
      <NotificationMenu
        notifications={noNotifs}
        authUser={mockUserFromResource}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
