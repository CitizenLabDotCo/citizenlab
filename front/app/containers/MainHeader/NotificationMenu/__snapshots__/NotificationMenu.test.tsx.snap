// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationMenu /> renders correctly when everything is there 1`] = `
<NotificationMenu__Container>
  <NotificationCount
    count={0}
    dropdownOpened={false}
    onClick={[Function]}
  />
  <Dropdown
    content={
      <InfiniteScroll
        element="div"
        getScrollParent={null}
        hasMore={false}
        initialLoad={true}
        isReverse={false}
        loadMore={[MockFunction]}
        loader={
          <NotificationMenu__LoadingContainer>
            <Spinner
              color="#666"
              size="32px"
              thickness="3px"
            />
          </NotificationMenu__LoadingContainer>
        }
        pageStart={0}
        ref={null}
        threshold={50}
        useCapture={false}
        useWindow={false}
      >
        <NotificationMenu__NotificationsWrapper>
          <Notification
            notification={
              Object {
                "attributes": Object {
                  "created_at": "2019-04-16T14:32:17.095Z",
                  "idea_title": Object {
                    "nl-BE": "Qui nihil veniam consequatur.",
                  },
                  "initiating_user_first_name": undefined,
                  "initiating_user_last_name": undefined,
                  "initiating_user_slug": undefined,
                  "read_at": null,
                  "reason_code": undefined,
                  "type": "admin_rights_received",
                },
                "id": "aminRights",
                "relationships": Object {
                  "comment": undefined,
                  "idea_status": undefined,
                  "initiating_user": Object {
                    "data": Object {
                      "id": "fe947bd3-9c67-40ed-96a3-9a12c4c73b46",
                      "type": "user",
                    },
                  },
                  "invite": undefined,
                  "post": undefined,
                  "project": undefined,
                },
                "type": "notification",
              }
            }
          />
        </NotificationMenu__NotificationsWrapper>
      </InfiniteScroll>
    }
    left="auto"
    maxHeight="320px"
    mobileLeft="auto"
    mobileMaxHeight="280px"
    mobileRight="-15px"
    mobileWidth="220px"
    onClickOutside={[Function]}
    opened={false}
    right="-5px"
    top="42px"
    width="300px"
  />
</NotificationMenu__Container>
`;

exports[`<NotificationMenu /> renders correctly when there is no notifs 1`] = `
<NotificationMenu__Container>
  <NotificationCount
    count={0}
    dropdownOpened={false}
    onClick={[Function]}
  />
  <Dropdown
    content={
      <InfiniteScroll
        element="div"
        getScrollParent={null}
        hasMore={false}
        initialLoad={true}
        isReverse={false}
        loadMore={[MockFunction]}
        loader={
          <NotificationMenu__LoadingContainer>
            <Spinner
              color="#666"
              size="32px"
              thickness="3px"
            />
          </NotificationMenu__LoadingContainer>
        }
        pageStart={0}
        ref={null}
        threshold={50}
        useCapture={false}
        useWindow={false}
      >
        <NotificationMenu__EmptyStateContainer>
          <NotificationMenu__EmptyStateImageWrapper>
            <NotificationMenu__EmptyStateImage
              alt=""
              role="presentation"
              src="test-file-stub"
            />
          </NotificationMenu__EmptyStateImageWrapper>
          <NotificationMenu__EmptyStateText>
            <FormattedMessage
              defaultMessage="You don't have any notifications yet"
              id="app.containers.NotificationMenu.noNotifications"
            />
          </NotificationMenu__EmptyStateText>
        </NotificationMenu__EmptyStateContainer>
      </InfiniteScroll>
    }
    left="auto"
    maxHeight="320px"
    mobileLeft="auto"
    mobileMaxHeight="280px"
    mobileRight="-15px"
    mobileWidth="220px"
    onClickOutside={[Function]}
    opened={false}
    right="-5px"
    top="42px"
    width="300px"
  />
</NotificationMenu__Container>
`;

exports[`<NotificationMenu /> renders correctly when there is no user 1`] = `""`;
